WoloxAndroidBootstrp/app/src/main/java/ar/com/wolox/android/example/ui/example/ExamplePresenter.kt

```diff
@@ -1,14 +1,41 @@
 package ar.com.wolox.android.example.ui.example
-
+import ar.com.wolox.android.example.model.LoginResponse
 import ar.com.wolox.android.example.utils.UserSession
-import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.core.presenter.CoroutineBasePresenter
+import ar.com.wolox.android.example.network.repository.PostRepository
+import ar.com.wolox.android.example.network.builder.networkRequest
+import kotlinx.coroutines.launch
+
 import javax.inject.Inject
+// CoroutineBasePresenter  PostRepository
+
+class ExamplePresenter @Inject constructor(private val userSession: UserSession, private val postRepository: PostRepository) : CoroutineBasePresenter<ExampleView>() {
+
+    fun onLoginButtonClicked(email: String, password: String) {
+
+        println("Cargando...")
+
+        view?.let { onLoginRequest(email, password) }
+    }
+
+    fun onLoginRequest(email: String, password: String) = launch {
+        println("Realizando request...")
+
+        networkRequest(postRepository.loginPostRepo(email, password)) {
+            onResponseSuccessful { response -> onSuccesResponse(response!!, email, password) }
```

Comment:
You can implement the onSuccessResponse method here, without the need to create that method.